controllers/discordactions.js

```diff
@@ -66,9 +67,25 @@ const createGroupRole = async (req, res) => {
 const getAllGroupRoles = async (req, res) => {
   try {
     const { groups } = await discordRolesModel.getAllGroupRoles();
+    const groupsWithMemberCount = await discordRolesModel.getNumberOfMemberForGroups(groups);
+    const groupCreatorIds = groupsWithMemberCount.reduce((ids, group) => {
+      ids.add(group.createdBy);
+      return ids;
+    }, new Set());
+    const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
+    const groupsWithUserDetails = groupsWithMemberCount.map((group) => {
+      const groupCreator = groupCreatorsDetails[group.createdBy];
+      return {
+        ...group,
+        firstName: groupCreator.first_name,
+        lastName: groupCreator.last_name,
```

Comment:
I dont think we need this here only user id is enough.

Reply:
I need this to show first and last names in UI (as expected in the issue). 
It's necessary for that. 

Instead of passing the user id and calling another API with all the userIDs to get all the details. It's better to add it here in a single API as the information is not much and this API gives all the collective details required for the UI to work.

Reply:
Resolving this for review. Let me know if you think otherwise and we can have a discussion on the API structure. 
Thank you for the review.